packages/wrangler/src/cloudchamber/ssh/ssh.ts

```diff
@@ -0,0 +1,402 @@
+import { readdir, stat, readFile } from "fs/promises";
+import { homedir, userInfo } from "os";
+import { exit } from "process";
+import {
+	error,
+	updateStatus,
+	log,
+	startSection,
+	endSection,
+	newline,
+	logRaw,
+	status,
+	success,
+} from "@cloudflare/cli";
+import { dim, brandColor } from "@cloudflare/cli/colors";
+import { inputPrompt, spinner } from "@cloudflare/cli/interactive";
+import { pollSSHKeysUntilCondition } from "../cli";
+import { SshPublicKeysService } from "../client";
+import {
+	handleFailure,
+	checkEverythingIsSet,
+	interactWithUser,
+} from "../common";
+import { wrap } from "../helpers/wrap";
+import { isInvalidPublicSSHKey } from "./validate";
+import type { Config } from "../../config";
+import type {
+	CommonYargsArgvJSON,
+	CommonYargsArgvSanitizedJSON,
+	StrictYargsOptionsToInterfaceJSON,
+} from "../../yargs-types";
+import type {
+	ListSSHPublicKeys,
+	SSHPublicKeyID,
+	SSHPublicKeyItem,
+} from "../client";
+
+function createSSHPublicKeyOptionalYargs(yargs: CommonYargsArgvJSON) {
+	return yargs
+		.option("name", {
+			type: "string",
+			describe:
+				"The alias to your ssh key, you can put a recognisable name for you here",
+		})
+		.option("public_key", {
+			type: "string",
+			describe:
+				"An SSH public key, you can specify either a path or the ssh key directly here",
+		});
+}
+
+/**
+ *  Tries to retrieve teh SSH key if sshKeyPath is a path to a ssh key.
+ *
+ *  If the parameter is a ssh key, it will return it.
+ *
+ *  On both paths of the code we will validate the SSH key.
+ */
+async function retrieveSSHKey(
+	sshKeyPath: string,
+	{ json }: { json: boolean } = { json: false }
+): Promise<string> {
+	try {
+		const file = (await readFile(sshKeyPath)).toString();
+		const reason = isInvalidPublicSSHKey(file);
+		if (reason !== null) {
+			if (!json) error(reason);
+			else console.log(JSON.stringify({ error: reason }, null, 4));
```

Comment:
This should use `logger.error()` (as should any other uses of `console.*`)

Reply:
Can we just do logger.log on places json only is expected?